libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -994,36 +1007,29 @@ void _IotMqtt_DecrementConnectionReferences( _mqttConnection_t * pMqttConnection
 IotMqttError_t IotMqtt_Init( void )
 {
     IotMqttError_t status = IOT_MQTT_SUCCESS;
+    uint32_t initializationStatus = Atomic_CompareAndSwap_u32( &_initCalled,
+                                                               MQTT_LIBRARY_INITIALIZED,
+                                                               MQTT_LIBRARY_UNINITIALIZED );
 
-    if( _initCalled == 0 )
+    if( initializationStatus != 0 )
```

Comment:
Should the `0` be `MQTT_LIBRARY_UNINITIALIZED` as it is below, or is `MQTT_LIBRARY_UNINITIALIZED` only meant to be assigned to `_initCalled`?

Reply:
`initializationStatus` is the return value from `Atomic_CompareAndSwap_u32()`, so it will be `1` if the CAS was successful and `0` otherwise.  This test is correct, the one below is not.